### DISCUSSION TITLE
Adding overlays to an UI

### DISCUSSION THREAD

**Discussion by 4JX** [emilk/egui#1615] [Q&A]
Reference pic:
![image](https://user-images.githubusercontent.com/79868816/167683497-db8f949b-9ae1-4424-a835-a2cb1506fa47.png)

I'm looking for a way to get these types of overlays on egui. `Area` seems to be my best fit for them, but I feel using a whole area to paint something like the fade-out black overlay is somewhat overkill.

Is there a more primitive version I can use for the fade overlay (since all it needs to be is a rectangle that covers the window area)? And would a non-draggable `Area` be suited for the job in the UI containing parts or is there something else I can look into?

**Comment by emilk** [comment]
`Area` is your only bet right now.
